### PR TITLE
new bf partitioning code for path repair as well as tests and documentation

### PR DESCRIPTION
**Pull request by jadenvc**

### CODE REVIEW

File: docs/replanning_tutorial.rst [hunk 1]
```diff
@@ -0,0 +1,67 @@
+Relinking
```
Comment (k2shah) [inline docs/replanning_tutorial.rst:1]: can you add these as sub pages to 
`tutorial.rst`? right now this page and the other one won't show up

File: wadl/lib/route.py [hunk 2]
```diff
@@ -167,6 +175,40 @@ def DistGPS(gps0, gps1, alt0=0, alt1=0):
...
+    def approx(self, a, b):
+        # accepts 2 utm points and return true if they are approximately the same value
+        if abs(a[0]-b[0]) < 0.1 and abs(a[1]-b[1]) < 0.1:
+            return True
+        return False
+
+    def setMaze(self, maze):
+        self.maze = maze
+
+    def unstreamline(self, metaGraph, i):
```
Comment (k2shah) [inline wadl/lib/route.py:187]: i see what you did here, but man we need a way better way to manage all these abstractions of routes

File: wadl/survey.py [hunk 3]
```diff
@@ -229,6 +231,64 @@ def relink(self, routeParameters, write=True, showPlot=False):
         # call shutdowns and free stuff
         self.close()
 
+
```
Comment (k2shah) [inline wadl/survey.py:234]: what is "completePct" ?

**Review by k2shah — CHANGES_REQUESTED** [review]
some super minor stuff here, but mostly looks good. I'll prob tweet the documentation pages once this is merged in. Can you also add your relinking script to the `examples` dir? like `stanford_relink.py`?

File: wadl/survey.py [hunk 4]
```diff
@@ -211,6 +211,8 @@ def relink(self, routeParameters, write=True, showPlot=False):
             #curSolver.metaGraph = copy.deepcopy(curSolver.metaGraphCopy) # dont need?
 
             maze.routeSet.routeParameters = routeParameters
```
[the remark this answers is not in the document]
Comment (jadenvc) [reply wadl/survey.py:213]: done

File: wadl/survey.py (hunk 3 above)
Comment (jadenvc) [reply wadl/survey.py:234]: percent of routes that will not complete (its just for simulating partial completion)

**Pull request closed without merging**